### PR TITLE
MONIT-21413: Add support for the new event format in java-lib

### PR DESCRIPTION
**Pull request by yzheqing**

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/dto/Event.java [hunk 1]
```diff
@@ -3,20 +3,15 @@
...
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
-import java.util.stream.Collectors;
 
-import static com.wavefront.common.SerializerUtils.appendQuoted;
-import static com.wavefront.common.SerializerUtils.appendTagMap;
-import static com.wavefront.common.SerializerUtils.appendTags;
+import static com.wavefront.common.SerializerUtils.*;
```
Comment (srujann) [inline java-lib/src/main/java/com/wavefront/dto/Event.java:14]: per our code style, we don't use wildcard imports.

File: java-lib/src/main/java/com/wavefront/ingester/EventIngesterFormatter.java [hunk 2]
```diff
@@ -39,40 +40,58 @@ public ReportEvent drive(String input, Supplier<String> defaultHostNameSupplier,
...
+    if (event.getHosts() == null && defaultHostNameSupplier != null) {
+      event.setHosts(ImmutableList.of(defaultHostNameSupplier.get()));
+    }
+    if (event.getHosts() == null) {
+      throw new IllegalArgumentException("hosts can't be null: " + input);
+    }
+    if (event.getGroup() == null) {
+      event.setGroup(defaultGroup);
+    }
+    // TODO: what to set for default eventId?
```
Comment (srujann) [inline java-lib/src/main/java/com/wavefront/ingester/EventIngesterFormatter.java:91]: setting random UUID looks fine. Remove the TODO.

File: java-lib/src/test/java/com/wavefront/ingester/EventDecoderTest.java [hunk 3]
```diff
@@ -49,47 +49,39 @@ public void testDecodeEventWithIncorrectKvPairsThrows() {
...
-    assertArrayEquals(new String[] {"eventtag1", "eventtag2"}, out.get(0).getTags().toArray());
-    assertEquals(4, out.get(0).getAnnotations().size());
-    assertNull(out.get(0).getDimensions());
-    assertEquals("Really long description with a line break here: \n end of description",
-        out.get(0).getAnnotations().get("description"));
-    assertEquals("deployment-event", out.get(0).getAnnotations().get("type"));
-    assertEquals("INFO", out.get(0).getAnnotations().get("severity"));
-    assertEquals("value", out.get(0).getAnnotations().get("somerandomannotation"));
+    assertEquals(7, out.get(0).getAnnotations().size());
+    // TODO: shall we put description into detail?
```
Comment (srujann) [inline java-lib/src/test/java/com/wavefront/ingester/EventDecoderTest.java:66]: yes, the description should go into `details` field, make the change for this.

**Review by srujann — COMMENTED** [review]
A couple of comments related to TODO.

**Review by srujann — APPROVED** [review]

**Review by haosong — APPROVED** [review]

**Pull request merged**